sdk/python/sawtooth_sdk/processor/core.py

```diff
@@ -33,65 +44,174 @@
 class TransactionProcessor(object):
     def __init__(self, url):
         self._stream = Stream(url)
+        self._url = url
         self._handlers = []
-        self._stop = False
 
     def add_handler(self, handler):
+        """Add a transaction family handler
+        :param handler:
+        """
         self._handlers.append(handler)
 
-    def start(self):
-        futures = []
-        for handler in self._handlers:
-            for version in handler.family_versions:
-                for encoding in handler.encodings:
-                    future = self._stream.send(
-                        message_type=Message.TP_REGISTER_REQUEST,
-                        content=TpRegisterRequest(
-                            family=handler.family_name,
-                            version=version,
-                            encoding=encoding,
-                            namespaces=handler.namespaces).SerializeToString())
-                    futures.append(future)
+    def _find_handler(self, header):
+        """Find a handler for a particular (family_name,
+        familiy_versions, payload_encoding)
+        :param header transaction_pb2.TransactionHeader:
+        :return: handler
+        """
+        return list(filter(lambda h: header.family_name == h.family_name and
+                           header.family_version in h.family_versions and
+                           header.payload_encoding in h.encodings,
+                           self._handlers))[0]
 
-        for future in futures:
-            LOGGER.debug("future result: %s", repr(future.result()))
+    def _register_requests(self):
+        """Returns all of the TpRegisterRequests for handlers
 
-        while True:
-            if self._stop:
-                break
-            msg = self._stream.receive()
-            LOGGER.debug(
-                'received message of type: %s',
-                Message.MessageType.Name(msg.message_type))
+        :note: itertools.chain.from_iterable flattens the nested lists
```

Comment:
Is this note useful as API doc, or should it be a non-API doc comment within the method?